### ISSUE TITLE
ISSUE: FedEx delivery update email not scanned

### ISSUE THREAD

**Issue by JvdMaat** [moralmunky/Home-Assistant-Mail-And-Packages#622]
**Describe the bug**
Getting a FedEx package delivered today. But due to circumstances, it was supposed to be delivered tomorrow. But overnight was updated to be delivered today, so FedEx never sent me the initial email to let me know I would be getting a package, and only sent the delivery update email, but the email wasn't scanned by this plugin.

**Environment (please complete the following information):**
 - OS: Home Assistant OS 7.1
 - Type of device: RPi4
 - Home Assistant version: 2021.12.10
 - Component version: 0.3.3-2

I can provide the full FedEx email, just don't want to post it publicly here. Short info:
From: FedEx Delivery Manager <TrackingUpdates@fedex.com>
Subject: FedEx Shipment <trackid>: Your package delivery has been updated
Body looks very similar to a regular shipment notification.

**Comment by JvdMaat** [comment]
I am not getting those. But I have signed up for every possible email on my delivery manager profile:
![image](https://user-images.githubusercontent.com/90044149/151215299-f961cb87-7bfe-43ec-ae89-8a8d94461162.png)
The only email I received for this delivery was the aforementioned updated delivery notification. I am guessing this was because when the tracking number was created, the shipper used a more distant address, and FedEx figured it would be delivered tomorrow. But when they received the package yesterday at a closer location, they updated the delivery to today overnight. So maybe the day of delivery emails go out first, and I wasn't part of that, and then the package delivery date was updated to today, and I only received the Delivery update notification?
Can the integration also parse the delivery update notifications and, if the date is today, add that to the deliveries, since FedEx sees fit not to send out an out for delivery email in those cases?

**Comment by firstof9** [comment]
Does it happen to say FedEx SmartPost?

**Comment by JvdMaat** [comment]
LOL, nope. This was a FedEx end to end:
SERVICE TYPE:		FedEx Home Delivery

I can forward you an email copy if you'd like so you can see the full tracking history.

**Comment by firstof9** [comment]
I've never seen a FedEx package that didn't get a "Your package is now out for delivery today" email unless it was SmartPost, then it's a 50/50 chance between FedEx and USPS delivering it.

**Comment by JvdMaat** [comment]
Yeah, I have no idea. I checked my SPAM folder and there's nothing. There is just the delivery update email, that's it. Like I said, I wonder if it got missed on the "out for delivery" deadline and had its deliver date updated afterwards, so that email did not go out.
There was definitely a last minute delivery date change. I purchased this yesterday with promise of next day delivery. But the shipper sent out an automated notification at 22:30 saying "sorry for the delay, it will arrive a day later", but then FedEx (in its only email to me) sent the delivery notification update saying it would arrive today. So it was an overnight delivery date update within the FedEx system.
Having said that, the FedEx email also came at 22:30 last night. So no idea why they would not also send out the out for delivery email. My account is enabled for it. (Technically I should have gotten three emails. Since I have checked notification on a package addressed to me, as well as day before and day of delivery.)

And now that I'm saying that out loud, since the delivery update came in last night, the integration would not be able to scan that for delivery today, as the email was from yesterday. So cases like these (however common or rare they are) would be unsupportable.

**Comment by firstof9** [comment]
_Normally_ the out for delivery message comes when they scan the packages when loading the trucks.

**Comment by JvdMaat** [comment]
Maybe I live near a bad FedEx station where they don't do that? Let me open up a case with them and see if they can shed light on this.

**Comment by JvdMaat** [comment]
FedEx customer service had no answers. Said I should have received it, and "they will look into it". And call back if it happens again.

**Comment by firstof9** [comment]
At least they followed up with you, unlike my local USPS lol

**Comment by JvdMaat** [comment]
Hah! Don't get me started on USPS. Moved here in June, and I **STILL** don't have informed delivery.
This house was split into a duplex in 2006, but they shared the same +4 ZIP, They fixed that by giving our side a new +4 in November. But that just means that our tenant gets all our informed delivery info now since they have the old +4.
(To be fair, it doesn't help that USPS thinks our house numbers are 123A and 123B, but we use 123, and the tenant uses 123 Apt B. Not sure how to get this problem resolved. All online databases just recognize the address as 123, but with the old +4 ZIP)
And yeah, that took from June to November to get the +4 issue identified and fixed. From USPS.com to the local post office, nobody knew what was wrong until I finally got to the right person who identified and fixed that in a day)
